hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/ECKeyOutputStream.java

```diff
@@ -548,7 +553,7 @@ private void addStripeToQueue(ECChunkBuffers stripe) throws IOException {
   private boolean flushStripeFromQueue() throws IOException {
     try {
       ECChunkBuffers stripe = ecStripeQueue.take();
-      while (!(stripe instanceof EOFDummyStripe)) {
+      while (!closing && !(stripe instanceof EOFDummyStripe)) {
```

Comment:
There might be some stripes buffered in `ecStripeQueue`, and `EOFDummyStripe` is used as signal of there is no more stripes left. Is it intended to drop the buffered stripes when `closing`?

Reply:
It guess it can only be closing=true at this point if there was a previous write exception that got thrown back to the caller. At that point, the caller needs to call close(). If they try to write again, they should see errors as the stream is "closing".

However I do wonder if we should check for and flush queued stripes in the close() method when closing = true.